packages/dds/map/src/mapKernel.ts

```diff
@@ -312,26 +325,28 @@ export class MapKernel {
      */
     public delete(key: string): boolean {
         // Delete the key locally first.
-        const successfullyRemoved = this.deleteCore(key, true);
+        const previousValue = this.deleteCore(key, true);
 
         // If we are not attached, don't submit the op.
         if (!this.isAttached()) {
-            return successfullyRemoved;
+            return previousValue !== undefined;
         }
 
         const op: IMapDeleteOperation = {
             key,
             type: "delete",
         };
-        this.submitMapKeyMessage(op);
+        this.submitMapKeyMessage(op, previousValue ?? "none");
```

Comment:
This is weird: why submit an op if the key didn't exist? Seems like we should check for that and do nothing in that case.

Reply:
I'll leave it.